### DISCUSSION TITLE
I don't understand the difference between microphone input and speakers

### DISCUSSION THREAD

**Discussion by dln949** [marin-m/SongRec#147] [Q&A] [answered]
In SongRec, I see two options:  

Turn the microphone recognition on or off; and, 

Recognize from my speakers instead of microphone

This greatly confuses me.  

If I have microphone recognition on, then I think, Well, SongRec is working based on the sound coming out of the speakers that the microphone is "hearing".  Is that correct?  If that is correct, then isn't SongRec recognizing the song from the speakers?  

But if I turn microphone recognition off, and check Recognize from my speakers instead of the microphone, I think to myself, How is that different, because in the first option it was relying on the sound coming out of the speakers?  

Do you understand my confusion?

**Comment by marin-m** [comment, marked as answer]
"From my speakers" is a simplified way to say "from the monitor device of my default audio output", and "from my microphones" to say "from my default audio input".

Where your default audio output is likely your speakers (or headset, ...) and your audio input, if you have, is likely a microphone.

e.g., if you play YouTube or Spotify, recognizing songs from your speakers will apply recognition directly over the sound going out of your computer's speakers.

If you are in an outdoor environment, recognizing songs from your microphone (if you have one) will apply recognition over what you are hearing around you.

Your microphone (if you have one) does not mirror directly the output going towards your speakers' sound interface. There is attenuation due to physical phenomenon, the quality of sound is at least degraded, or the sound may even not be present when your microphone is muted while your speakers are on, ...
